src/blocks/calendar.scss

```diff
@@ -0,0 +1,48 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  padding: 0;
+  margin: 0 auto;
+  max-width: $calendar-width;
+  list-style: none;
+
+  @each $length, $days in $month-lengths {
+    &.calendar--length-#{$length} {
+      .calendar__day:nth-child(n + #{$days + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  &__day {
+    background-color: #eee;
+    height: $day-size;
+    width: $day-size;
+    border: 1px black solid;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    box-sizing: border-box;
+
+    &:hover {
+      transition: transform 500ms;
+      background-color: #ffbfcb;
+      transform: translate(0, -20px);
+    }
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        content: '#{$i}';
+      }
+    }
+  }
+
+  @each $day, $index in $start-days {
+    &.calendar--start-#{$day} {
+      .calendar__day:first-child {
+        margin-left: calc($day-size * $index + $gap * $index);
+      }
+    }
+  }
```

Comment:
The variable `$start-days` is used here but not defined within this file. Ensure it is defined elsewhere in your project or imported if defined in another file.